cmd/xl-v1-list-objects-heal.go

```diff
@@ -205,3 +206,237 @@ func (xl xlObjects) ListObjectsHeal(bucket, prefix, marker, delimiter string, ma
 	// Return error at the end.
 	return ListObjectsInfo{}, toObjectErr(err, bucket, prefix)
 }
+
+// ListUploadsHeal - lists ongoing multipart uploads that require
+// healing in one or more disks.
+func (xl xlObjects) ListUploadsHeal(bucket, prefix, marker, uploadIDMarker,
+	delimiter string, maxUploads int) (ListMultipartsInfo, error) {
+
+	if err := checkListObjsArgs(bucket, prefix, marker, delimiter, xl); err != nil {
+		return ListMultipartsInfo{}, err
+	}
+
+	switch {
+	// With max uploads of zero we have reached eof, return right here.
+	case maxUploads == 0:
+		return ListMultipartsInfo{}, nil
+	case maxUploads > maxUploadsList, maxUploads < 0:
+		return ListMultipartsInfo{}, InvalidMaxUpload{maxUploads}
+	}
+
```

Comment:
The check is moved to `admin-handlers.go` and is the only place we validate the query parameters. `maxUploads == 0` in the updated PR would return `InvalidMaxUploads` error.